moveit_ros/planning_interface/move_group_interface/include/moveit/move_group_interface/move_group_interface.h

```diff
@@ -400,7 +399,7 @@ class MoveGroupInterface
 
       If these values are out of bounds then false is returned BUT THE VALUES
       ARE STILL SET AS THE GOAL. */
-  bool setJointValueTarget(const sensor_msgs::JointState& state);
+  MOVEIT_DEPRECATED bool setJointValueTarget(const sensor_msgs::JointState& state);
```

Comment:
Why not just change the implementation to match `setJointValueTarget(const std::map<std::string, double>& variable_values)`? It would probably make sense to use a `setJointValueTarget(const std::vector<std::string>& variable_names, const std::vector<double>& variable_names)` for that.

Reply:
There is already a `setJointValueTarget(const std::map<std::string, double>& variable_values)`.
I'm not sure about your intention.

Reply:
My intention is to keep support for setting target states from robot trajectory waypoints directly (from both message and class types). I don't get why `setJointValueTarget(const sensor_msgs::JointState& state)` should be deprecated only because there can be joints that are not member of the group. IMO this function should just ignore non-member joints just like `setJointValueTarget(const std::map<std::string, double>& variable_values)` does.
Alternatively I would suggest adding a `setJointValueTarget(const std::vector<std::string>& variable_names, const std::vector<double>& variable_names)` function that is more convenient to use than the map.

Reply:
> My intention is to keep support for setting target states from robot trajectory waypoints directly

Makes sense. Please feel free to augment to this PR branch.